src/test/java/BaseTest.java

```diff
@@ -4,17 +4,25 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.*;
 
 import java.time.Duration;
 import java.util.UUID;
 
 
 public class BaseTest {
+   //remember if declaring non-primitive data types, must assign values-strings, arrays, classes
     public static WebDriver driver = null;
     public static String url = null;
+    public static WebDriverWait wait = null;
 
+    Actions action = new Actions(driver);
```

Comment:
Please finish the homework! You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/213907184-e72cd25a-7c45-4712-b16e-0fdab0c98f45.png)

Hint 

`public static Actions action;`

```
@BeforeMethod
    @Parameters({"BaseURL"})
    public static void launchBrowser(String BaseURL) {
        LoginTests.driver = new ChromeDriver();
        action = new Actions(driver);
        LoginTests.driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        url = BaseURL;
        driver.get(url);
        wait = new WebDriverWait (LoginTests.driver, Duration.ofSeconds(10));

    }
```